src/components/WalletSwitcher.tsx

```diff
@@ -71,15 +71,16 @@ const WalletSwitcher = () => {
     if (!switchToWalletName) return
 
     setIsPasswordModalOpen(false)
-    unlockWallet(
-      switchToWalletName,
+    unlockWallet({
+      event: 'switch',
```

Comment:
It makes our state easier to reason about if we know exactly what triggered the wallet unlock (whether it was the user switching wallets after already having one unlocked, or if it was a normal login when no other wallet was unlocked. That's why I introduced the event type (see other comment).

Reply:
Yes, it makes total sense.